pkg/controllers/clusterresourceplacement/placement_controller.go

```diff
@@ -333,3 +339,116 @@ func (r *Reconciler) updatePlacementAppliedCondition(placement *fleetv1alpha1.Cl
 		}
 	}
 }
+
+// handleUpdate handles the create/update clusterResourcePlacement event.
+// It creates corresponding clusterPolicySnapshot and clusterResourceSnapshot if needed and updates the status based on
+// clusterPolicySnapshot status and work status.
+func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1.ClusterResourcePlacement) (ctrl.Result, error) {
+	crpKObj := klog.KObj(crp)
+	policyHash, err := generatePolicyHash(crp.Spec.Policy)
+	if err != nil {
+		klog.ErrorS(err, "Failed to generate policy hash of crp", "clusterResourcePlacement", crpKObj)
+		return ctrl.Result{}, err
+	}
+
+	clusterPolicySnapshotList := &fleetv1.ClusterPolicySnapShotList{}
+	if err := r.Client.List(ctx, clusterPolicySnapshotList, client.MatchingLabels{fleetv1.CRPTrackingLabel: crp.Name}); err != nil {
```

Comment:
sg, my original idea is to reduce the API server call. Since it should be rare case there is no active snapshot, it's reasonable  to list with active label first. will address it.